docs/developer-docs/integrations/sns/get-sns/local-testing.md

```diff
@@ -1,74 +1,150 @@
-Remove yourself, as well as any other developers,
-from the list of controllers that the dapp canister(s) have.
+Make a call to the SNS wasm modules canister on the local NNS to request that an SNS is installed.
+Installing the SNS has some preconditions:
+
+1. The SNS configuration (created in the [previous step](./preparation.md)) has to be named `sns.yml` and has to be placed in your project's root directory.
+2. Your [cycles wallet](../../../build//project-setup/cycles-wallet.md) has to be added to the whitelist of principals that are allowed to create SNSes. This is a temporary measure to make sure that no one launches an SNS before the code is ready. In the long-term, this condition will be removed. 
+3. Your cycles wallet contains enough cycles to create an SNS. Currently, this cost is 50T cycles.
+
+The SNS is new and may still have significant bugs.
+To prevent huge numbers of projects giving control of their dapps to SNSs before the SNS has been tested in production, 
+access to the SNS is limited. The decision who gets to be one of the brave first projects to launch an SNS and be added to this whitelist is decided by the NNS community by proposal.
+Later, when the SNS has a solid track record in production, this whitelist will be dropped.
+For local development, you can add your wallet to the whitelist using the following command and identity `developer-identity`:
+
+``` bash
+# This command assumes that you are using an unencrypted identity and that you are using the identity you want to deploy the SNS with.
+$(dfx cache show)/ic-admin --secret-key-pem ~/.config/dfx/identity/$(dfx identity whoami)/identity.pem --nns-url "https://localhost:$(dfx info replica-port)" propose-to-update-sns-deploy-whitelist --added-principals "$(dfx identity get-wallet)" --proposer "$DEVELOPER_NEURON_ID" --proposal-title "Let me SNS!" --summary "I am friendly."
+```
+
+Creating an SNS currently uses up to 250T cycles, and your wallet needs to supply those.
+On a local deployment, you can add any number of cycles to any canister.
+To add 2345T cycles to your wallet, run the following command:
+
+``` bash
+dfx ledger fabricate-cycles --canister $(dfx identity get-wallet) --t 2345
+```
+
+And to check the new balance, run `dfx wallet balance`.
+
+Now that everything is set up, you can deploy the sns using `dfx sns deploy`.
+This will deploy the SNS and print some canister IDs.
+Add the printed canister IDs manually to the file `.dfx/local/canister_ids.json`.
+Then make sure you can access them with:
+
+```bash
+dfx canister id sns_root
+dfx canister id sns_governance
+dfx canister id sns_ledger
+dfx canister id sns_swap
+```
+
+And to see how many cycles the deployment cost, run `dfx wallet balance` again.
+
+#### 4. Remove all controllers other than the SNS from the dapp canister(s)
+Remove yourself, as well as any other developers, from the list of controllers that the dapp canister(s) have.
```

Comment:
```suggestion
Remove yourself, as well as any other developers, from the list of controllers that the dapp canister(s) have and add the SNS root canister as their controllers.
```

( the command does both, right? i.e., overwrite the current list with SNS root? I would add this as I was first confused "if we remove the current controllers here, how can we then add the right controller?")